### PR TITLE
Migrations -- tests

### PR DESCRIPTION
**Pull request by radex**
- add missing tests for migrations
- fix bugs discovered:
  - fix migrations failing on Loki
  - fix migrations being re-applied on next relaunch on Loki
  - fix default value not being set for new columns on SQLite
- simplify common tests

### CODE REVIEW

**Comment by radex** [comment]
@kokusGr CR please. Note that CI fails because this doesn't include Android native changes (Leon's branch) yet

File: src/adapters/__tests__/commonTests.js [hunk 1]
```diff
@@ -451,15 +452,162 @@ export default () => [
...
+
+      expect(await adapter.count(checkTaskColumn('test_string', ''))).toBe(2)
+      expect(await adapter.count(checkTaskColumn('test_string_optional', null))).toBe(2)
+      expect(await adapter.count(checkTaskColumn('test_number', 0))).toBe(2)
+      expect(await adapter.count(checkTaskColumn('test_number_optional', null))).toBe(2)
+      expect(await adapter.count(checkTaskColumn('test_boolean', false))).toBe(2)
+      expect(await adapter.count(checkTaskColumn('test_boolean_optional', null))).toBe(2)
+
+      // check I can use new table and columns
+      adapter.batch([
```
Comment (kokusGr) [inline src/adapters/__tests__/commonTests.js]: Don't need `await` here?

File: src/adapters/__tests__/commonTests.js [hunk 2]
```diff
@@ -451,15 +452,162 @@ export default () => [
...
+        [
+          'create',
+          new MockTask(
+            {},
+            { id: 't3', test_string: 'hey', test_number: 2, test_boolean_optional: true },
+          ),
+        ],
+      ])
+
+      // check that out-of-range migration was not executed
```
Comment (kokusGr) [inline src/adapters/__tests__/commonTests.js]: Shouldn't out of range migration throw an error?

**Comment by kokusGr** [comment]
Looks ok, only 2 questions :)

File: src/adapters/__tests__/commonTests.js (hunk 2 above)
Comment (radex) [reply src/adapters/__tests__/commonTests.js]: Not sure what you're asking — the point is that database we're testing is already at version 3, so migrations from v2 to v3 must not be executed (and we're testing that the migration was not in fact applied)

File: src/adapters/__tests__/commonTests.js (hunk 1 above)
Comment (radex) [reply src/adapters/__tests__/commonTests.js]: Oops! Fixed in #81

File: src/adapters/__tests__/commonTests.js (hunk 2 above)
Comment (kokusGr) [reply src/adapters/__tests__/commonTests.js]: OK, my bad :)

**Pull request merged**